dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java

```diff
@@ -394,12 +394,15 @@ public Result queryProjectListPaging(User loginUser, Integer pageSize, Integer p
         List<User> userList = userMapper.selectByIds(projectList.stream()
                 .map(Project::getUserId).distinct().collect(Collectors.toList()));
         Map<Integer, String> userMap = userList.stream().collect(Collectors.toMap(User::getId, User::getUserName));
-        List<ProjectWorkflowDefinitionCount> projectWorkflowDefinitionCountList =
-                workflowDefinitionMapper.queryProjectWorkflowDefinitionCountByProjectCodes(
-                        projectList.stream().map(Project::getCode).distinct().collect(Collectors.toList()));
-        Map<Long, Integer> projectWorkflowDefinitionCountMap = projectWorkflowDefinitionCountList.stream()
-                .collect(Collectors.toMap(ProjectWorkflowDefinitionCount::getProjectCode,
-                        ProjectWorkflowDefinitionCount::getCount));
+        List<Long> projectCodes = projectList.stream().map(Project::getCode).distinct().collect(Collectors.toList());
+        Map<Long, Integer> projectWorkflowDefinitionCountMap = Collections.emptyMap();
+        if (CollectionUtils.isNotEmpty(projectCodes)) {
+            projectWorkflowDefinitionCountMap = workflowDefinitionMapper
+                    .queryProjectWorkflowDefinitionCountByProjectCodes(projectCodes)
+                    .stream()
+                    .collect(Collectors.toMap(ProjectWorkflowDefinitionCount::getProjectCode,
+                            ProjectWorkflowDefinitionCount::getCount));
+        }
```

Comment:
If the `projectList` is empty, then we can return early at line 389, then the logic would be clearer?